pkg/controller/job/backoff_utils.go

```diff
@@ -183,12 +184,29 @@ func getFinishedTime(p *v1.Pod) time.Time {
 }
 
 func getFinishTimeFromContainers(p *v1.Pod) *time.Time {
-	var finishTime *time.Time
-	for _, containerState := range p.Status.ContainerStatuses {
-		if containerState.State.Terminated == nil {
-			return nil
+	finishTime := latestFinishTime(nil, p.Status.ContainerStatuses, nil)
+	// We need to check InitContainerStatuses here also,
```

Comment:
the code doesn't look risky, but we should be consistent and check the feature gate to skip the additional calculations when the feature is disabled

Reply:
yes, most of cases it is. However there is almost no validation on consistency of statuses: https://github.com/kubernetes/kubernetes/pull/124918

Reply:
are you saying that the code i this PR is more risky than it appears?

Reply:
the only risk is for the crazy scenarios when some sort of virtual kubelet was setting some bad values for init containers and it used to be ignored before. This is why I suggest only add a logic for sidecars.